### ISSUE TITLE
Symfony 6 support

### ISSUE THREAD

**Issue by chris-doehring** [dogado-group/json-api-server#5]
Symfony 6 will be [released in November 2021](https://symfony.com/releases/6.0). `symfony/validator` is the only relevant dependency in this package and the support for the new version should be added before it's release.

**Comment by chris-doehring** [comment]
To have a valid symfony validator instance, that does not require `doctrine/annotations`, we need `symfony/validator:^5.2` and the following initialization:
```php
// for php 8 annotations
(new ValidatorBuilder())>enableAnnotationMapping(true)->getValidator();

// with annotation support
(new ValidatorBuilder())>enableAnnotationMapping()->getValidator();
```

Under Symfony 6, it look's like this:
```php
// for php 8 annotations
(new ValidatorBuilder())>enableAnnotationMapping()->getValidator();

// with annotation support
(new ValidatorBuilder())>enableAnnotationMapping()->addDefaultDoctrineAnnotationReader()->getValidator();
```

Further details regarding the attribute support can be found in their [release blog post](https://symfony.com/blog/new-in-symfony-5-2-constraints-as-php-attributes). [However, there is a limitation to some validation constraints, that will propably be solved with php 8.1.](https://github.com/symfony/symfony/issues/38503)

All variations of the Symfony validator in version 5 and 6 are still supported in the json-api-server version 1.x and the future 2.x release. In PR #7 I've added a phpunit test build to check the Symfony 6 compatibility. `symfony/validator:^6.0` will be officially supported in this package as soon as there is a `v6.0.0` release.

**Comment by chris-doehring** [comment]
In the [`v2.0.0`](https://github.com/dogado-group/json-api-server/releases/tag/v2.0.0) release I've added some preperations for Symfony 6. I tested the compatibility with `6.0.x-dev` and found no issues. I created a [new workflow for GitHub actions](https://github.com/dogado-group/json-api-server/actions/workflows/phpunit-symfony6.yml) which will check the compatibility to the unit tests on every commit.

The next step is to wait for the offical `symfony/validator:v6.0.0` release and change the local version constraint to `^5.2 || ^6.0`.
